plugins/modules/docker_swarm_service.py

```diff
@@ -2551,7 +2551,9 @@ def _detect_update_config_failure_action_rollback(client):
     rollback_config_failure_action = (
         (client.module.params['update_config'] or {}).get('failure_action')
     )
-    update_failure_action = client.module.params['update_failure_action']
+    update_failure_action = (
+        client.module.params.get('update_failure_action')
```

Comment:
I'm pretty sure that this is the wrong fix. A module option of this name no longer exists. I suspect we forgot to remove this part when removing some deprecated options.

It's probably best to remove this line and change the next two to
```
     return rollback_config_failure_action == 'rollback'
```

CC @dariko @jwitko @hannseman @WojciechowskiPiotr